### DISCUSSION TITLE
Can this work on a VPS ?

### DISCUSSION THREAD

**Discussion by VuzzyM** [trinib/Adguard-Wireguard-Unbound-Cloudflare#14] [F.A.Q]
On vps ununtu is work?

**Comment by trinib** [comment]
Yes. I tried everything with this set up on `Amazon EC2,` and `Vultr`, both worked well( [more VPS services](https://github.com/dalisoft/awesome-hosting#vps) ). I had followed this [guide ](https://github.com/isystem-io/wireguard-docs/tree/master/docs/en) which was to set up wireguard on a VPS(aws), you can get some reference there. (Vultr is easy to set up) </br>Mainly what ever VPS your using you will have to get a public ip4 address and open needed ports, and sometimes in linux VM you need to allow all outside connections in [UFW](https://help.ubuntu.com/community/UFW) firewall rules. For example 
```
sudo ufw allow in on (name of your Ethernet interface)
```
`ip maddress` to show name of network interfaces

Or allow specific public IP only:
```
sudo ufw allow from 211.112.36.0/254 
```
_(0 is replaced with the last digits)_
</br>

With AdGuard Home installed only, accessible from anywhere with ssl certificate from let's encrypt that [auto renews](https://github.com/acmesh-official/acme.sh#12-how-to-renew-the-certs) every 60 days using [acme.sh](https://github.com/acmesh-official/acme.sh) :

![vps](https://user-images.githubusercontent.com/18756975/167276898-461cb9b2-5ce7-4a53-ad98-6b1a48641003.gif)


With WireGuard Vpn:
![aws](https://user-images.githubusercontent.com/18756975/161938545-2403e72e-c037-4745-bcc5-2766ad07e514.png)
